### DISCUSSION TITLE
Flask 标语（web development, one drop at a time）翻译投票与征集

### DISCUSSION THREAD

**Discussion by greyli** [greyli/flask-docs-zh#37] [Vote]
Flask logo 里的「web development, one drop at a time」是否翻译，如何翻译，欢迎大家投票决定（**点评论左边的向上箭头**），也欢迎给出你的翻译备选项（一条评论给出一条翻译）。

附上 QQ 群里一位翻译专业的同学提供的建议：

> 你好。研究生读了翻译专业的我想提个建议：「Web development, one drop at a time.」最好不要直译。作为一个 motto，「一次一滴」不够上口，不容易让人明白它跟 web 开发有什么关系，甚至还会让人产生不好的联想。我建议用保留部分原意方法来「创译」（transcreate），或者完全不拘泥于原文重新写一句口号/motto。
> 
> 首先是三个前提：
> 1.「one drop at a time」是一句药品（滴剂）说明书上的用法描述，或者医嘱，也就是「（每天三次，）每次一滴」。
> 2.「Flask」是随身携带的酒瓶，也可以是盛放各种化学试剂的「烧瓶」，或者直接就是「瓶」。
> 3. 没有理解错的话，我认为 Flask 用这句口号是为了突出它的轻量和可扩展，「一次只用一滴」vs「一次需要喝下一整瓶」。
> 
> 那么这里就有两个意象：酒，和药。如果要迁就酒的意象，那么我觉得只能放弃「one drop at a time」，例如：
> Web 醇酿，尽在此「瓶」。
> 此处「醇酿」也可换成「美酒」「佳酿」等等。这么译的好处是强化了「flask」的意象，甚至澄清了这个词乃至 Flask logo 的含义。
> 
> 如果要迁就「药」，那么可以译为：
> Web 灵药，点滴病除。
> 这样最符合原文的含义，也不失中文的美感。
> 
> 但这两种译法的缺点是，第一次看到的读者可能会摸不着头脑。（也许把上面两句的前半部分都换成「Web开发」，会好一点？）
> 
> 除此之外，比较稳妥但没有特色的译法，可以有：
> Web开发，起于点滴。
> Web开发，水滴石穿。
> Web世界，尽在此「瓶」。
> 
> 在此抛砖引玉一下。

**Comment by greyli** [comment]
web development, one drop at a time（不译）

**Comment by greyli** [comment]
Web 开发，一次一滴

**Reply by frostming** [reply]
Web 开发，每次一滴

**Reply by sungeer** [reply]
web开发的点点滴滴

**Comment by greyli** [comment]
Web 醇酿，尽在此「瓶」

**Comment by greyli** [comment]
Web 佳酿，尽在此「瓶」

**Comment by greyli** [comment]
Web 美酒，尽在此「瓶」

**Comment by greyli** [comment]
Web 灵药，点滴病除

**Comment by greyli** [comment]
Web 开发，起于点滴

**Comment by greyli** [comment]
Web 开发，水滴石穿

**Comment by greyli** [comment]
Web 世界，尽在此「瓶」

**Comment by aiguoli** [comment]
web开发，滴水不漏

**Comment by aiguoli** [comment]
web开发，一蹴而就

**Comment by frostming** [comment]
web开发，涓滴而成

**Comment by abersheeran** [comment]
我把它翻译成“任取随用的 Web 框架”。在断句上，现代中文断句是含义中断才会使用逗号进行断句，在原文中，“one drop at a time” 显然是对 “Web development” 的形容，按照现代中文表达的习惯，应该翻译成 “xxxx 的 xxxx”。如果追求断句相同，那就得寻古了。古文确实有许多这种类似的表达结构，比如这个也可以翻译成“Web 开发，起自微毫”。

**Comment by zodiacx8** [comment]
“聚沙成塔"怎么样？

**Reply by greyli** [reply]
有一个 Python Web 框架叫 Pyramid :p

**Reply by yougaUsth** [reply]
积水成渊如何？ 也符合‘’one drop at a time ‘’的寓意

**Reply by greyli** [reply]
我的理解是这里的 one drop at a time 是指按需取用，而不是积少成多。

**Reply by zodiacx8** [reply]
~~嗯，我觉得Flask应该是指的烧瓶。喝酒的时候不会考虑 one drop at a time 吧😂。至于药品，强调one drop at a time的一般都是猛药，和Flask相关度不高。这里的 one drop at a time指的应该是烧瓶可以准确地称量液体，也许可以用”不差毫厘“？~~
简单查了一下，Flask这里应该是酒瓶。Flask官网的图标用的是可以用于盛放Whiskey的[Horn Flask](https://youtu.be/d7j4CU_cNuk)。"one drop at a time" 应该是与下文的”micro“ feature 对应的。
> The “micro” in microframework means Flask aims to keep the core simple but extensible.  
> -from [Flask doc](https://flask.palletsprojects.com/en/2.0.x/foreword/#what-does-micro-mean)

简单粗暴点，就翻译成”简单灵活的Web开发框架“？没有calture background可能理解不了Whiskey相关的内容。
ps:原来官网上的图标不是chili pepper😂

**Reply by greyli** [reply]
> 这里的 one drop at a time指的应该是烧瓶可以准确地称量液体

这里的「酒」指的是 Python 的 WSGI 标准（经常被读作 Whiskey），一次一滴指的是可以按需使用。

**Comment by greyli** [comment]
援引 [V2EX 上的评论](https://www.v2ex.com/t/779715#r_10567513)，也可以考虑译作「渐进式 Web 框架」。

**Comment by ghost** [comment]
基于以上的讨论，我倾向于“不译”或“Web 开发，起自微毫“

**Comment by leonsama** [comment]
[官方文档对微框架的解释](https://flask.palletsprojects.com/en/2.0.x/foreword/#what-does-micro-mean)

我认为 `one drop at a time` 表达的是 `对症下药` 的意思。结合官网对 Flask 的定位，Flask是一个极其灵活的渐进式框架。Flask 给开发者提供了高自由度的选择，开发者可以“按需取用” Flask 的功能。

翻译标语应该在易于理解的情况下突出主体特点。考虑官网的设计，我认为翻译成 `强大而精，按需构建` 比较好

**Comment by yinian1992** [comment]
参看 http://mitsuhiko.pocoo.org/flask-pycon-2011.pdf
Flask 显然不是 Bottle，而是烧瓶。

**Reply by greyli** [reply]
Flask 是 [drinking horn](https://www.google.com.hk/search?q=drinking+horn&newwindow=1&safe=strict&sxsrf=ALeKk018GrWEST_GRxQiqbmcaMxlvJZz2A:1623246673280&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj1ut-K2YrxAhUEEqYKHV8fArMQ_AUoAXoECAEQAw&biw=1279&bih=1320)。

**Reply by yinian1992** [reply]
按词源 flask 就是古英语/中世纪拉丁语中的“瓶”，作者是玩了个文字游戏，与 Bottle 框架区分开，所以不要直接用“瓶”这个概念了吧。

**Reply by greyli** [reply]
同意，「瓶」更容易让人想到 Bottle。

**Comment by greyli** [comment]
根据 [Flask 用户调查问卷投票结果](https://greyli.com/flask-report-2021/)，翻译暂定为「web 开发，起于点滴」。感谢各位给出的建议和投票！
